src/sss_client/pam_message.c

```diff
@@ -158,6 +159,9 @@ int pack_message_v3(struct pam_items *pi, size_t *size, uint8_t **buffer)
     rp += add_uint32_t_item(SSS_PAM_ITEM_CLI_PID, (uint32_t) pi->cli_pid,
                             &buf[rp]);
 
+    rp += add_uint32_t_item(SSS_PAM_ITEM_CHILD_PID, (uint32_t) pi->child_pid,
```

Comment:
Hi,

I'd like to suggest underline that this is an optional field by only setting it if `pi->child_pid > 0` and also add a comment in `pam_parse_in_data_v2()` that this is optional data. It does not matter much on a single system but in case we really start running SSSD in a container and share sockets with other containers we might have to be more strict on those interfaces.

bye,
Sumit

Reply:
```diff
diff --git a/src/responder/pam/pamsrv_cmd.c b/src/responder/pam/pamsrv_cmd.c
index e1e5f7b73..ab622c2a2 100644
--- a/src/responder/pam/pamsrv_cmd.c
+++ b/src/responder/pam/pamsrv_cmd.c
@@ -349,6 +349,7 @@ static int pam_parse_in_data_v2(struct pam_data *pd,
                     if (ret != EOK) return ret;
                     break;
                 case SSS_PAM_ITEM_CHILD_PID:
+                    /* This is optional. */
                     ret = extract_uint32_t(&pd->child_pid, size,
                                            body, blen, &c);
                     if (ret != EOK) return ret;
diff --git a/src/sss_client/pam_message.c b/src/sss_client/pam_message.c
index 9bf61fa6b..17033aa29 100644
--- a/src/sss_client/pam_message.c
+++ b/src/sss_client/pam_message.c
@@ -159,8 +159,10 @@ int pack_message_v3(struct pam_items *pi, size_t *size, uint8_t **buffer)
     rp += add_uint32_t_item(SSS_PAM_ITEM_CLI_PID, (uint32_t) pi->cli_pid,
                             &buf[rp]);

-    rp += add_uint32_t_item(SSS_PAM_ITEM_CHILD_PID, (uint32_t) pi->child_pid,
-                            &buf[rp]);
+    if (pi->child_pid > 0) {
+        rp += add_uint32_t_item(SSS_PAM_ITEM_CHILD_PID,
+                                (uint32_t) pi->child_pid, &buf[rp]);
+    }

     rp += add_authtok_item(SSS_PAM_ITEM_AUTHTOK, pi->pam_authtok_type,
                            pi->pam_authtok, pi->pam_authtok_size, &buf[rp]);

```